apps/web/src/components/NftFailureFallback/NftFailureBoundary.tsx

```diff
@@ -1,34 +1,163 @@
-import { createElement, useMemo } from 'react';
+import { createElement, useEffect, useMemo } from 'react';
+import { useRelayEnvironment } from 'react-relay';
+import { fetchQuery, graphql } from 'relay-runtime';
+import styled from 'styled-components';
 
 import { useNftErrorContext } from '~/contexts/NftErrorContext';
+import { NftFailureBoundaryPollerQuery } from '~/generated/NftFailureBoundaryPollerQuery.graphql';
+import { useNftRetry } from '~/hooks/useNftRetry';
 import {
   ReportingErrorBoundary,
   ReportingErrorBoundaryProps,
 } from '~/shared/errors/ReportingErrorBoundary';
+import colors from '~/shared/theme/colors';
+
+import { VStack } from '../core/Spacer/Stack';
+import {
+  NftFailureFallback,
+  NftFailureFallbackProps,
+  NftFallbackLabel,
+} from './NftFailureFallback';
 
 type Props = {
   tokenId: string;
-} & ReportingErrorBoundaryProps;
+  fallbackSize?: NftFailureFallbackProps['size'];
+  fallback?: ReportingErrorBoundaryProps['fallback'];
+  loadingFallback?: ReportingErrorBoundaryProps['fallback'];
+} & Omit<ReportingErrorBoundaryProps, 'fallback'>;
+
+export function NftFailureBoundary({
+  tokenId,
+  fallbackSize = 'medium',
+  fallback = <NftFailureFallback tokenId={tokenId} size={fallbackSize} />,
+  loadingFallback = <NftLoadingFallback size={fallbackSize} />,
+  ...rest
+}: Props) {
+  const { handleNftError, retryKey } = useNftRetry({
+    tokenId,
+  });
+
+  const { tokens, clearTokenFailureState, markTokenAsPolling } = useNftErrorContext();
+
+  const additionalTags = useMemo(() => ({ tokenId }), [tokenId]);
+
+  const token = tokens[tokenId];
+
+  const relayEnvironment = useRelayEnvironment();
+
+  useEffect(
```

Comment:
- default error fallbacks
- default loading fallbacks
- automatic retries if syncing
- automatic error reporting
- automatic resolution if syncing token completes processing

Reply:
this is sickkkk